### PR TITLE
cmd/candid: rename put-agent to create-agent

### PR DESCRIPTION
**Pull request by rogpeppe**
This command now only creates agents - it does not update them.

Also enable environment-variable-based agent-authentication
for all candid commands.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/48/

File: cmd/candid/internal/admincmd/command.go [hunk 1]
```diff
@@ -86,15 +86,24 @@ func (c *candidCommand) Client(ctxt *cmd.Context) (*candidclient.Client, error)
...
+		authInfo = ai
+	} else if ai, err := agent.AuthInfoFromEnvironment(); err == nil {
+		authInfo = ai
+	} else if errgo.Cause(err) != agent.ErrNoAuthInfo {
+		return nil, errgo.Mask(err)
+	}
+	if authInfo != nil {
+		// Agent authentication has been specified, so we probably don't
+		// want to use existing cookies (which might be logged in as a different
+		// user) or to fall back to interactive authentication.
```
Comment (mhilton) [inline cmd/candid/internal/admincmd/command.go]: I'm not sure how true that is, what if the auth info in the environment is for a different service?

Comment (rogpeppe) [reply cmd/candid/internal/admincmd/command.go]: I think it's more true than not - we can add another flag if need be.

**Comment by rogpeppe** [comment]
@fabricematrat I'll address the docs issue in the next PR, where I'm moving the common flags into the supercommand.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**